ex02.exs

```diff
@@ -1,6 +1,27 @@
 
 defmodule Ex02 do
 
+  ### API ###
+
+  def new_counter(initial_value \\ 0),        do: make_counter(initial_value, :standard)
+  def new_global_counter(initial_value \\ 0), do: make_counter(initial_value, :global)
+
+  def next_value(counter), do: counter |> Agent.get_and_update( &{ &1, &1+1 } )
+
+  def global_next_value(), do: next_value(__MODULE__)
+
+  ### Implementation ###
+
```

Comment:
but I'm not too sure about this particular chunk o' code.

That's not to imply I dislike it. It means I don't know whether I like it or not. 

My confusion is over the `make_counter` function, which basically does nothing except extract the pid from the tuple returned by start_link.

I wonder if something like this mike reduce duplication:

~~~ elixir
@named { name: __MODULE__ }
@unnamed {}

def new_counter(initial_value \\ 0),        do: make_counter(initial_value, @unnamed)
def new_global_counter(initial_value \\ 0), do: make_counter(initial_value, @named)

defp make_counter(value, options) do
  { :ok, counter } = Agent.start_link(fn -> value end, options)
  counter
end
~~~